### PR TITLE
Fix #81 - Initial Welcome Experience

### LINKED ISSUE #81: Handling initial [No Name] Buffer
This issue is more of a brain dump regarding what I've found about this buffer since it has proven to be non-trivial to handle. This issue relating to it first came up in #72. Currently once #76 is merged, it will show its modified status correctly and close when the cross is clicked.

If it is navigated away from on start, it is discarded. If it is modified then a user tries to return to it, currently it silently fails in Oni2.

I've found that actually this is the same behavior as terminal neovim which is actually errorring with `E499: Empty file name for '%' or '#', only works with ":p:h"`. Essentially we're not allowed to navigate to a file without a name, which this file does not have nor does using its id work.

It can still be closed though. I think as what it does is default vim behavior this is fine for now, It  makes me think though that even at this early stage it might be worth considering a splash screen and only opening buffers a user has chosen to open. 

Either way we aren't doing anything that differs from neovim, but it highlights the fact that we aren't rendering errors which is confusing but should be fixed once we pick up the `ext_messages` change in nvim 0.4.0

### PR DESCRIPTION
**Pull request by bryphe**
This adds support for custom 'renderers' for buffers - the idea is that we can skip out on rendering the `<EditorSurface />` in some cases.

This will be useful for:
- Welcome experience
- Terminal
- Image rendering

And potentially future displays (markdown, pdf).

For now, though, this is about adding a welcome experience - @glennsl and I discussed this a bit:
image

There's a few goals I have for this immediate PR:
- [x] Animate / polish the logo
- [x] Show some of the basic keybindings for getting around:
  - [x] QuickOpen (Control/Command+P)
  - [x] Command Palette (Control/Command + Shift + P)
  - [x] Vim Ex mode (`:`)
- [x] Add support for overriding tab name w/ custom renderer 

Then, as next steps, I'd like to add to the screen:
- Build info
- Link to our documentation 
- 'Sneak mode' entry point (Control+G)
- Recent files 
- Recent workspaces

One other idea for the welcome screen was to add some very small "mini-tutorials" to learn the basic keybindings - like `h`/`j`/`k`/`l` - or some quick vim tips

Fixes #81

### CODE REVIEW

File: src/UI/WelcomeView.re [hunk 1]
```diff
@@ -13,84 +13,120 @@ open Oni_Core;
...
+        width(300),
+      ];
+
+    let commandText = (~theme: Theme.t, ~fontFile, ~fontSize as fs) =>
+      Style.[
+        fontFamily(fontFile),
+        fontSize(fs),
+        backgroundColor(theme.background),
+        color(theme.editorForeground),
+      ];
```
Comment (glennsl) [inline src/UI/WelcomeView.re]: If you remove `Style.` here in favor of `open Style;` at the beginning of the `Styles` module, you don't have to rename `~fontSize`. Instead you can qualify the style property: `Style.fontSize(fontSize)`. Much less awkward IMO.

Comment (bryphe) [reply src/UI/WelcomeView.re]: Nice!

**Comment by bryphe** [comment]
I'll bring this in for now, but still needs some refinement + polish.

**Comment by glennsl** [comment]
Looks very nice! Neat fade! The buffer is still called `untitled` in the Ctrl+Tab menu though.

**Comment by bryphe** [comment]
Thanks @glennsl ! Animation API is great 😉 

> The buffer is still called untitled in the Ctrl+Tab menu though.

Oh, good catch, I guess we should have a more central location to override the buffer name. Logged #1110 to track

**Pull request merged**